### PR TITLE
Adding new OCP4 role for RHMI 2.x (Integreatly)

### PR DESCRIPTION
**Pull request by pmccarthy**
##### SUMMARY
Adding new base template OCP4 role for managing Red Hat Managed Integration (Integreatly) installations. This role does not do anything yet but will act as the foundation for future development which will be progressed over the coming days/weeks

Equivalent OCP3 role can be found [here](https://github.com/redhat-cop/agnosticd/tree/development/ansible/roles/ocp-workload-integreatly)

##### ISSUE TYPE
- New role Pull Request

##### COMPONENT NAME
Red Hat Managed Integration (Integreatly)

### CODE REVIEW

**Comment by pmccarthy** [comment]
@NonyNo3lle @fridim Hey guys, perhaps you could have a quick look at this PR please? Or if not, direct me to someone who might be able to take a look? It's just a basic role that's not tied to any catalog item right now so should be safe to merge back. Thanks!

**Comment by wkulhanek** [comment]
@pmccarthy I would suggest checking out the example workload that we recently updated: ocp-workload-example and starting by making a copy of that one.

We introduced a much cleaner way of passing variables into a workload using Ansible dictionaries. Now since I don't really know Integreatly I don't know if this is necessary - or if Integreatly always deploys the same. But it would make things a bit more encapsulated and less likely to collide variables with other parts of AgnosticD.

We also started using the meta/main.yml to put information about the roles including owners. So usually we don't have a n OWNERS file. Not that it will hurt.

Feel free to ping me anytime if you want more explanation than what's in the readme etc. of that role.

**Comment by pmccarthy** [comment]
@wkulhanek thanks for the update. I'll go ahead and make the appropriate changes based on the example role you've referenced.

File: ansible/roles/ocp4-workload-integreatly/tasks/post_workload.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+
+
+# Leave these as the last tasks in the playbook
+# ---------------------------------------------
+
+# For deployment onto a dedicated cluster (as part of the
+# cluster deployment) set workload_shared_deployment to False
+# This is the default so it does not have to be set explicitely
+- name: pre_workload tasks complete
```
Comment (austincunningham) [inline ansible/roles/ocp4-workload-integreatly/tasks/post_workload.yml:14]: Shouldn't this be
```suggestion
- name: post_workload tasks complete
```

File: ansible/roles/ocp4-workload-integreatly/tasks/post_workload.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  debug:
+    msg: "Post-Workload tasks completed successfully."
+  when:
+  - not silent|bool
+  - not workload_shared_deployment|d(False)
+
+# For RHPDS deployment (onto a shared cluster) set
+# workload_shared_deployment to True
+# (in the deploy script or AgnosticV configuration)
+- name: pre_workload tasks complete
```
Comment (austincunningham) [inline ansible/roles/ocp4-workload-integreatly/tasks/post_workload.yml:24]: Same here ^^

**Comment by pmccarthy** [comment]
@wkulhanek I've just made updates to reflect the latest role configurations if you wouldn't mind taking a look please?

Also, couple of questions:
1) I wasn't sure what information is mandatory in the `meta/main.yml` file. I left out the `license` section as I'm not sure it's applicable?
2) In the README, is the agnosticv deployment section required? I'm not sure this is something that will be used in this role?

Thanks

File: ansible/roles/ocp4-workload-integreatly/tasks/post_workload.yml (hunk 1 above)
Comment (pmccarthy) [reply ansible/roles/ocp4-workload-integreatly/tasks/post_workload.yml:14]: @austincunningham - good spot, thanks

File: ansible/roles/ocp4-workload-integreatly/tasks/post_workload.yml (hunk 2 above)
Comment (wkulhanek) [reply ansible/roles/ocp4-workload-integreatly/tasks/post_workload.yml]: Oh. Yeah. I'll fix that in the example. The important part (the message) is correct. It's just the name of the task. Thanks for spotting. Duh.

**Comment by wkulhanek** [comment]
@pmccarthy Thanks. I'll take a look. As for AgnosticV: I'm 99.9% sure sure it will be required. All new configs use AgnosticV to set the variables rather than our legacy shell scripts.

Once again it will depend if the workload can be customized or if it always uses the same settings.

Some of mine set defaults (e.g. for cli tool versions) that then can be overwritten using AgnosticV. That way we don't have to go and change AgnosticD and add new tags. We just update the config.

The meta/main is just whatever you put there. :-) It's just a standard place (and ansible-galaxy creates a template for it).

**Pull request merged**